django/template/defaulttags.py

```diff
@@ -1167,6 +1167,54 @@ def now(parser, token):
     return NowNode(format_string, asvar)
 
 
+@register.simple_tag(takes_context=True)
+def query_string(context, query_dict=None, **kwargs):
+    """
+    Add, remove, and change parameters of the request's `QueryDict`,
+    and return the result as a querystring.
+
+    For example::
+
+        {% query_string foo=3 %}
+
+    To remove a key::
+
+        {% query_string foo=None %}
+
+    To use with pagination::
+
+        {% query_string page=page_obj.next_page_number %}
+
+    A custom `QueryDict` can also be used:
+
+        {% query_string my_query_dict foo=3 %}
+    """
+    if query_dict:
+        query_dict = query_dict.copy()
+    else:
+        query_dict = context.request.GET.copy()
+    keys_to_remove = []
+    for key, value in kwargs.items():
+        if value is None:
+            keys_to_remove.append(key)
+        try:
+            iter(value)
+        except TypeError:
+            query_dict[key] = value
+        else:
+            if isinstance(value, str):
+                query_dict[key] = value
+            else:
+                query_dict.setlist(key, value)
+    for key in keys_to_remove:
+        if key in query_dict:
+            del query_dict[key]
+    query_string = query_dict.urlencode()
+    if not query_string:
+        return ""
```

Comment:
Can we swap these two lines? There is no need to encode a blank string.
```suggestion
    if not query_string:
        return ""
    query_string = query_dict.urlencode()
```